core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/natlang/JSGFGenerator.java

```diff
@@ -0,0 +1,206 @@
+public class JSGFGenerator {
+	
+	/**
+	 * a list of all rules
+	 */
+	List<String> rules;
+	
+	private String name;
+	private String wakeup;
+	private String sleep;
+	private String shutdown;
+	
+	/**
+	 * constructor, generates a JSGF file
+	 * @param name the name of the grammar file
+	 * @param wakeup the wakeup command
+	 * @param sleep the sleep command
+	 * @param shutdown the shutdown command
+	 */
+	public JSGFGenerator(String name, String wakeup, String sleep, String shutdown) {
+		this.name = name;
+		this.wakeup = wakeup;
+		this.sleep = sleep;
+		this.shutdown = shutdown;
+		
+		this.rules = new ArrayList<>();
+	}
+	
+	/**
+	 * generates a whole JSGF grammar file content
+	 * including previously added rules via addRule(..)
+	 * 
+	 * just put the returning string in a new file
+	 * 
+	 * @return the JSGF grammar string, including headers
+	 */
+	public String generateGrammarFileString() {
+		StringBuilder grammar = new StringBuilder();
+		
+		grammar.append("#JSGF V1.0;\n" + "\n" + "/**\n" + " * JSGF Grammar \n" + " */\n" + "\n");
+
+		grammar.append("grammar " + this.name + ";\n");
+		grammar.append("public <wakeup> = ( " + this.wakeup + " );\n");
+		grammar.append("public <sleep> = ( " + this.sleep + " );\n");
+		grammar.append("public <shutdown> = ( " + this.shutdown + " );\n");
+
+		grammar.append("\n//pre defined rules \n");
+
+		// pre defined rules
+		grammar.append("<digit> = (one | two | three | four | five | six | seven |"
+				+ "nine | ten | eleven | twelve | thirteen | fourteen | fifteen | "
+				+ "sixteen | seventeen | eighteen | nineteen | twenty | thirty | forty | "
+				+ "fifty | sixty  | seventy | eighty | ninety | and )+; \n");
+
+		grammar.append("\n//custom rules \n");
+		
+		
+		for(String rule : this.rules) {
+			grammar.append(rule);
+		}
+		
+		return grammar.toString();
+	}
+	
+	/**
+	 * generates a valid public JSGF Rule from an AGFNode
+	 * the generated rule is added to an internal list of rules, if 
+	 * generateGrammarFileString() is called added rules are included
+	 * @param node root node
+	 * @param ruleName the name of the rule
+	 * @return the rule as String
+	 */
+	public String addRule(AGFNode node, String ruleName) {
+		StringBuilder b = new StringBuilder();
+		b.append("public <" + ruleName + "> = ");
+		String rule = handleNode(b, node) + ";";
+		
+		this.rules.add(rule);
+		return rule;
+	}
+
+	/**
+	 * traverses the tree
+	 * 
+	 * @param b the string builder
+	 * @param node the current root node
+	 * @return the current string representation of the rule
+	 */
+	private String handleNode(StringBuilder b, AGFNode node) {
+		if(node != null) {
+			switch(node.getType()) {
+			case WORD:
+				b.append(" " + node.getContent());
+				break;
+			case RULE:
+				b.append(" <digit>");
+				break;
+			case ORG:
+				b.append(" (");
+				for(int i = 0; i < node.getChilds().size(); i++) {
+					handleNode(b, node.getChilds().get(i));
+					if(i!=node.getChilds().size()-1) {
+						b.append("|");
```

Comment:
maybe it makes sense to pack this into a method. I think it is to much code for a case statement